tests/install_vm.py

```diff
@@ -38,32 +45,87 @@ def parse_args():
+    )
 
     return parser.parse_args()
 
+
 def main():
     data = parse_args()
-    data.kickstart = "https://raw.githubusercontent.com/ComplianceAsCode/content/master/tests/kickstarts/rhel_centos_7.cfg"
-    data.disk_path = os.path.join(data.disk_dir, data.domain) + ".qcow"
+    data.disk_path = os.path.join(data.disk_dir, data.domain) + ".qcow2"
+    data.ks_basename = os.path.basename(data.kickstart)
 
     if data.distro == "fedora":
         data.variant = "fedora27" # this is for support in RHEL7, where fedora28 is not known yet
-        data.url = "https://download.fedoraproject.org/pub/fedora/linux/releases/28/Everything/x86_64/os"
+        if not data.url:
+            data.url = "https://download.fedoraproject.org/pub/fedora/linux/releases/29/Everything/x86_64/os"
     elif data.distro == "centos7":
         data.variant = "centos7"
-        data.url = "http://mirror.centos.org/centos/7/os/x86_64"
+        if not data.url:
+            data.url = "http://mirror.centos.org/centos/7/os/x86_64"
+    elif data.distro == "rhel7":
+        data.variant = "rhel7.0"
+    elif data.distro == "rhel8":
+        data.variant = "rhel8.0"
+
+    if not data.ssh_pubkey:
+        username = ""
+        try:
+            username = os.environ["SUDO_USER"]
+        except KeyError:
+            pass
+        home_dir = os.path.expanduser('~' + username)
+        data.ssh_pubkey = home_dir + "/.ssh/id_rsa.pub"
+    if not os.path.isfile(data.ssh_pubkey):
+        sys.stderr.write("Error: SSH public key not found at {0}\n".format(data.ssh_pubkey))
+        sys.stderr.write("You can use the `--ssh-pubkey` to specify which key should be used.\n")
+        return 1
+    with open(data.ssh_pubkey) as f:
+        pub_key = f.readline().rstrip()
+    print("Using SSH public key from file: {0}".format(data.ssh_pubkey))
 
-    command = 'virt-install -n {domain} -r {ram} --vcpus={cpu} --os-variant={variant} --accelerate --disk path={disk_path},size=12 -x "inst.ks={kickstart}" --location {url}'.format(**data.__dict__)
+    tmp_kickstart = "/tmp/" + data.ks_basename
+    with open(data.kickstart) as infile, open(tmp_kickstart, "w") as outfile:
+        old_content = infile.read()
+        new_content = old_content.replace("&&HOST_PUBLIC_KEY&&", pub_key)
+        outfile.write(new_content)
+    data.kickstart = tmp_kickstart
+    print("Using kickstart file: {0}".format(data.kickstart))
+
+    # The kernel option 'net.ifnames=0' is used to disable predictable network
+    # interface names, for more details see:
+    # https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/
+    command = 'virt-install --name={domain} --memory={ram} --vcpus={cpu} --os-variant={variant} --hvm --accelerate --network=default --disk path={disk_path},size=20,format=qcow2 --initrd-inject={kickstart} --extra-args="inst.ks=file:/{ks_basename} ksdevice=eth0 net.ifnames=0" --wait=-1 --vnc --location={url}'.format(**data.__dict__)
```

Comment:
In Fedora (maybe in RHEL too), if instead of `--network=default`, `--network bridge=virbr0` is specified. The VM will be created in the user session of libvirtd (i.e. `qemu:///session`), and `sudo` won't be required to run tests.

Reply:
I will update the script to handle both `qemu:///system` and `qemu:///session` hypervisors and set the default to be the `qemu:///session`.